extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -281,6 +283,16 @@ private void checkProperConfigAccessInProvider(AnnotationInstance instance) {
         }
     }
 
+    private void checkProperConstructorInProvider(AnnotationInstance i) {
+        ClassInfo targetClass = i.target().asClass();
+        if (!targetClass.hasNoArgsConstructor()
+                || targetClass.methods().stream().filter(m -> m.name().equals("<init>")).count() > 1) {
+            LOGGER.warn(
+                    "Classes annotated with @Provider should have a single, no-argument constructor. Offending class is "
```

Comment:
Shouldn't we mention what happens if it does not have such a constructor?

Reply:
Makes sense. I'll update the message.

Reply:
Fixed